### ISSUE TITLE
SA1515 (Single-line comment must be preceded by blank line) should not be reported for single line comment directly following a case or default statement

### ISSUE THREAD

**Issue by vweijsters** [DotNetAnalyzers/StyleCopAnalyzers#878]
The following code will produce SA1515, but should not:

```C#
switch (someValue)
{
    case 1:
        // This should not produce SA1515
        return;
    default:
        break;
}
```

**Comment by sharwell** [comment]
This was a big factor in disabling the rule for now (I meant to create an issue for it but ran out of time). Thanks for updating it so quickly. :smile:

**Comment by AArnott** [comment]
Can we re-open this? It still repros with StyleCop.Analyzers 1.0.0

**Comment by sharwell** [comment]
@AArnott I believe that would be a separate issue
